uploader.go

```diff
@@ -35,14 +53,27 @@ func (u *Uploader) Offset() int64 {
 
 // Upload uploads the entire body to the server.
 func (u *Uploader) Upload() error {
+	info := UploadInfo{Offset: u.offset, TotalSize: u.upload.size, Percentage: 0}
 	for u.offset < u.upload.size && !u.aborted {
+		info.update(u.offset)
+
+		for _, c := range u.infoSubs {
```

Comment:
This will decrease performance if there's a lot of subscribers. We could send to a channel and somewhere else and this could run concurrently.